lib/tessel/setup.js

```diff
@@ -0,0 +1,139 @@
+//creates a .tessel folder with ssh keys in your home directory and uses those ssh keys to authorize you to push code to the USB-connected Tessel
+
+var osenv = require('osenv');
+var fs = require('fs-extra');
+var keygen = require('ssh-keygen');
+
+var Tessel = require('./tessel');
+var commands = require('./commands')
+
+var dir = osenv.home() + '/.tessel';
+var filename = 'id_rsa';
+var filepath = dir + '/' + filename;
+
+var tessel;
+
+Tessel.prototype.setupTessel = function (opts) {
+  var self = this;
+  // Check if local .tessel file has keypair, if not, put it there
+  setupLocal(function () {
+    // Authorize Tessel for SSH
+    authTessel(self);
+  });
+}
+
+// Make sure local computer is set up to authorize with Tessel
+function setupLocal (callback) {
+  // Check if keypair already in place
+  if(!(fs.existsSync(filepath) && fs.existsSync(filepath + '.pub'))) {
+    console.log('Creating public and private keys for Tessel authentication...');
+
+    // Generate SSH key
+    keygen({}, function(err, out){
+      if(err) return console.log('Something went wrong: '+err);
+      var privateKey = out.key;
+      var publicKey = out.pubKey;
+
+      // Make sure dir exists
+      fs.ensureDir(dir, function (err) {
+        err && console.log(err);
+
+        filesWritten = 0;
+
+        // Put SSH keys for Tessel in that folder
+        fs.writeFile(filepath + '.pub', publicKey, function (err) {
```

Comment:
Eventually, I recommend using the `async` module for executing multiple async functions and getting a single callback. It will be more consistent with the rest of the code.